hardware/invensense/60xx/mlsdk/mllite/dmpconfig.txt
major version    =  1
minor version    =  0
startAddr        =  0


sdk/changes.txt
Change log for Android SDK Tools.

Revision 22.2.1:
- Emergency fix to new project templates.

Revision 22.2.0:

Revision 22:

Revision 20:
- Ant build system:
- Ant build system:
   * the location of the SDK can now be set through the ANDROID_HOME
     environment variable (requires updating the project's build.xml file)
   * Fixed issues with compilation, deployment of test projects and running tests.
     Both separate test projects and self-contained app+test projects now work.
   * Aidl files from library projects are now in the import list.

Revision 19:
- Fix GPU rendering

Revision 18:
* Fix Ant issues where some jar libraries in libs/ are not picked in some
  cases
* New version of SdkController

Revision 17:
* Bug fixes.
* ProGuard updated to version 4.7. In addition to many new features,
  this will fix the "Conversion to Dalvik format failed with error 1"
  error some users have experienced.
* The default proguard.cfg file has been updated with better
  flags.
* Lint now checks Java code, and checks which previously performed
  pattern based searches in the Java code (such as the unused resource
  check) have been rewritten to use the more accurate Java parse
  trees.
* Lint supports library projects, so for example the unused resource
  check will properly handle resources declared in a library project
  and referenced in a downstream project.
* Lint warnings can be suppressed in Java code with the new
  @SuppressLint annotation, and in XML files with the new tools:
  namespace and ignore-attribute.
* New lint checks:
  * Find Android API calls that require a version of Android higher
    than the minimum supported version. You can use the new @TargetApi
    annotation to specify local overrides for conditionally loaded
    code.
  * Find object allocations within onDraw, onMeasure and onLayout
    calls
  * Find usages of HashMap<Integer,X> which can be replaced by the
    more efficient SparseArray, SparseIntArray or SparseBooleanArray
  * Find typos in XML namespace declarations, as well as unused custom
    namespace declarations and usages of custom namespaces in library
    projects which do not yet work
  * Find manifest files which declare more than one <uses-sdk>
    elements, or which fail to declare minSdkVersion or
    targetSdkVersion
  * Check dialog button order such that OK/Cancel are in the right
    order (depending on the target version of Android), that they are
    using the standard case ("OK", not "Ok" or "ok" etc), and that
    there aren't any Back buttons.
  * Check menus to ensure that they are using "ifRoom" instead of
    "always" for the showAsAction attribute (unless it's used
    sparingly)
  * Find hardcoded android:debuggable attributes on <application>
    elements. This can lead to accidentally leaving debug information
    in published applications.
  * Find errors in format strings. This includes inconsistencies
    between translated versions, as well as incompatible types
    declared in XML and supplied in the String.format call in Java.
  * Find incorrect casts in Java code where the cast is incompatible
    with the view type declared in XML layouts.
  * Find hardcoded references to "/sdcard" in Java code; these should
    be using Environment.getExternalStorageDirectory().getPath()
    instead
  * Find "import android.R" statements in Java code, which can lead to
    confusing compilation error messages
  * Find Java code which creates world-writeable files (which can lead
    to security problems)
  * Find references to private resources (which can lead to runtime
    errors on devices that do not provide the resource, since private
    resources may disappear any time, and may not be present on all
    platforms.)
  * Find invalid @id references (where for example a RelativeLayout
    references an id that does not exist anywhere). If no match is
    found, it looks for spelling mistakes and suggests other similar
    ids in the error message.
  * Finds method calls to java.lang.Math where a float variable is
    cast to a double and/or the return value is cast to a float, and
    suggests replacing it with the equivalent android.util.FloatMath
    method.
  * Finds calls to a getter on the same class where a field could be
    accessed instead.
  * Find XML files using a different encoding format than UTF-8, since
    this can lead to subtle bugs (and lint was fixed to properly use
    the specified encoding)
  * Find XML resource files that contain text which should not be
    there (such as in layouts). This can identify accidentally
    malformed documents which happen to be valid XML, such as a
    missing "<" tag opening character.
  * Find style references which lead to cycles, such as extending
    self.

Revision 16:
* New "lint" tool which scans Android project trees for potential
  problems such as missing translations, duplicate ids between layouts
  that include each other, using px instead of dp units, hardcoded
  strings, missing contentDescriptions, obsolete Proguard
  configuration files, etc. For a full list of available issues run
  "lint --show" (and see eclipse/changes.txt for further details).
* layoutopt was removed; its functionality is replaced by the new lint
  tool

Revision 14:

- Build performance improvements:
   * resource compilation and packaging now properly use dependency to only
     be executed if a resource changed.
   * Optimized resource compilation for projects with libraries. This should
     speed up this phase significantly for large projects with libraries.
   * PNG files that are optimized during resource packaging are now cached
     and only re-optimized if they changed instead of doing at every build.
- New library project mechanism:
   *

Revision 13:

Revision 12: (07/2011):
- The AVD manager and emulator can now use system images compiled for ARM v7 and
  x86 CPUs.

Revision 11 (05/2011):
- See eclipse/changes.txt for ADT related changes.

Revision 10 (02/2011):

hardware/ti/omap3/dspbridge/inc/cfg.h
/*
 *  Copyright 2001-2008 Texas Instruments - http://www.ti.com/
 * 
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

/*
 *  ======== cfg.h ========
 *  DSP-BIOS Bridge driver support functions for TI OMAP processors.
 *  Purpose:
 *      PM Configuration module.
 *
 *  Private Functions:
 *      CFG_Exit
 *      CFG_GetAutoStart
 *      CFG_GetCDVersion
 *      CFG_GetDevObject
 *      CFG_GetDSPResources
 *      CFG_GetExecFile
 *      CFG_GetHostResources
 *      CFG_GetObject
 *      CFG_GetPerfValue
 *      CFG_GetWMDFileName
 *      CFG_GetZLFile
 *      CFG_Init
 *      CFG_SetDevObject
 *      CFG_SetObject
 *
 *! Revision History:
 *! =================
 *! 26-Feb-2003 kc  Removed unused CFG fxns.
 *! 28-Aug-2001 jeh  Added CFG_GetLoaderName.
 *! 26-Jul-2000 rr:  Added CFG_GetDCDName to retrieve the DCD Dll name.
 *! 13-Jul-2000 rr:  Added CFG_GetObject & CFG_SetObject.
 *! 13-Jan-2000 rr:  CFG_Get/SetPrivateDword renamed to CFG_Get/SetDevObject.
 *!                  CFG_GetWinBRIDGEDir/Directory,CFG_GetSearchPath removed.
 *! 15-Jan-1998 cr:  Code review cleanup.
 *! 16-Aug-1997 cr:  Added explicit cdecl identifiers.
 *! 12-Dec-1996 gp:  Moved CFG_FindInSearchPath to CSP module.
 *! 13-Sep-1996 gp:  Added CFG_GetBoardName().
 *! 22-Jul-1996 gp:  Added CFG_GetTraceStr, to retrieve an initial GT trace.
 *! 26-Jun-1996 cr:  Added CFG_FindInSearchPath.
 *! 25-Jun-1996 cr:  Added CFG_GetWinSPOXDir.
 *! 17-Jun-1996 cr:  Added CFG_GetDevNode.
 *! 11-Jun-1996 cr:  Cleaned up for code review.
 *! 07-Jun-1996 cr:  Added CFG_GetExecFile and CFG_GetZLFileName functions.
 *! 04-Jun-1996 gp:  Added AutoStart regkey and accessor function.  Placed
 *!                  OUT parameters in accessor function param. lists at end.
 *! 29-May-1996 gp:  Moved DEV_HDEVNODE to here and renamed CFG_HDEVNODE.
 *! 22-May-1996 cr:  Added GetHostResources, GetDSPResources, and
 *!                  GetWMDFileName services.
 *! 18-May-1996 gp:  Created.
 */

#ifndef CFG_
#define CFG_

#ifdef __cplusplus
extern "C" {
#endif

#include <dspapi.h>
#include <cfgdefs.h>

/*
 *  ======== CFG_Exit ========
 *  Purpose:
 *      Discontinue usage of the CFG module.
 *  Parameters:
 *  Returns:
 *  Requires:
 *      CFG_Init() was previously called.
 *  Ensures:
 *      Resources acquired in CFG_Init() are freed.
 */
	extern VOID CFG_Exit();

/*
 *  ======== CFG_GetAutoStart ========
 *  Purpose:
 *      Retreive the autostart mask, if any, for this board.
 *  Parameters:
 *      hDevNode:       Handle to the DevNode who's WMD we are querying.
 *      pdwAutoStart:   Ptr to location for 32 bit autostart mask.
 *  Returns:
 *      DSP_SOK:                Success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_RESOURCENOTAVAIL: Unable to retreive resource.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:        *pdwAutoStart contains autostart mask for this devnode.
 */
	extern DSP_STATUS CFG_GetAutoStart(IN struct CFG_DEVNODE* hDevNode,
					   OUT DWORD * pdwAutoStart);

/*
 *  ======== CFG_GetCDVersion ========
 *  Purpose:
 *      Retrieves the version of the PM Class Driver.
 *  Parameters:
 *      pdwVersion: Ptr to DWORD to contain version number upon return.
 *  Returns:
 *      DSP_SOK:    Success.  pdwVersion contains Class Driver version in
 *                  the form: 0xAABBCCDD where AABB is Major version and
 *                  CCDD is Minor.
 *      DSP_EFAIL:  Failure.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    Success.
 *      else:       *pdwVersion is NULL.
 */
	extern DSP_STATUS CFG_GetCDVersion(OUT DWORD * pdwVersion);

/*
 *  ======== CFG_GetDevObject ========
 *  Purpose:
 *      Retrieve the Device Object handle for a given devnode.
 *  Parameters:
 *      hDevNode:       Platform's DevNode handle from which to retrieve value.
 *      pdwValue:       Ptr to location to store the value.
 *  Returns:
 *      DSP_SOK:                Success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_INVALIDPOINTER:   phDevObject is invalid.
 *      CFG_E_RESOURCENOTAVAIL: The resource is not available.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    *pdwValue is set to the retrieved DWORD.
 *      else:       *pdwValue is set to 0L.
 */
	extern DSP_STATUS CFG_GetDevObject(IN struct CFG_DEVNODE* hDevNode,
					   OUT DWORD * pdwValue);

/*
 *  ======== CFG_GetDSPResources ========
 *  Purpose:
 *      Get the DSP resources available to a given device.
 *  Parameters:
 *      hDevNode:       Handle to the DEVNODE who's resources we are querying.
 *      pDSPResTable:   Ptr to a location to store the DSP resource table.
 *  Returns:
 *      DSP_SOK:                On success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_RESOURCENOTAVAIL: The DSP Resource information is not
 *                              available
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    pDSPResTable points to a filled table of resources allocated
 *                  for the specified WMD.
 */
	extern DSP_STATUS CFG_GetDSPResources(IN struct CFG_DEVNODE* hDevNode,
					      OUT struct CFG_DSPRES * pDSPResTable);


/*
 *  ======== CFG_GetExecFile ========
 *  Purpose:
 *      Retreive the default executable, if any, for this board.
 *  Parameters:
 *      hDevNode:       Handle to the DevNode who's WMD we are querying.
 *      cBufSize:       Size of buffer.
 *      pstrExecFile:   Ptr to character buf to hold ExecFile.
 *  Returns:
 *      DSP_SOK:                Success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_INVALIDPOINTER:   pstrExecFile is invalid.
 *      CFG_E_RESOURCENOTAVAIL: The resource is not available.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    Not more than cBufSize bytes were copied into pstrExecFile,
 *                  and *pstrExecFile contains default executable for this 
 *                  devnode.
 */
	extern DSP_STATUS CFG_GetExecFile(IN struct CFG_DEVNODE* hDevNode,
					  IN ULONG cBufSize,
					  OUT PSTR pstrExecFile);

/*
 *  ======== CFG_GetHostResources ========
 *  Purpose:
 *      Get the Host PC allocated resources assigned to a given device.
 *  Parameters:
 *      hDevNode:       Handle to the DEVNODE who's resources we are querying.
 *      pHostResTable:  Ptr to a location to store the host resource table.
 *  Returns:
 *      DSP_SOK:                On success.
 *      CFG_E_INVALIDPOINTER:   pHostResTable is invalid.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_RESOURCENOTAVAIL: The resource is not available.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    pHostResTable points to a filled table of resources 
 *                  allocated for the specified WMD.
 *
 */
	extern DSP_STATUS CFG_GetHostResources(IN struct CFG_DEVNODE* hDevNode,
					       OUT struct CFG_HOSTRES * pHostResTable);

/*
 *  ======== CFG_GetObject ========
 *  Purpose:
 *      Retrieve the Driver Object handle From the Registry
 *  Parameters:
 *      pdwValue:   Ptr to location to store the value.
 *      dwType      Type of Object to Get
 *  Returns:
 *      DSP_SOK:    Success.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    *pdwValue is set to the retrieved DWORD(non-Zero).
 *      else:       *pdwValue is set to 0L.
 */
	extern DSP_STATUS CFG_GetObject(OUT DWORD * pdwValue, DWORD dwType);

/*
 *  ======== CFG_GetPerfValue ========
 *  Purpose:
 *      Retrieve a flag indicating whether PERF should log statistics for the
 *      PM class driver.
 *  Parameters:
 *      pfEnablePerf:   Location to store flag.  0 indicates the key was
 *                      not found, or had a zero value.  A nonzero value
 *                      means the key was found and had a nonzero value.
 *  Returns:
 *  Requires:
 *      pfEnablePerf != NULL;
 *  Ensures:
 */
	extern VOID CFG_GetPerfValue(OUT bool * pfEnablePerf);

/*
 *  ======== CFG_GetWMDFileName ========
 *  Purpose:
 *    Get the mini-driver file name for a given device.
 *  Parameters:
 *      hDevNode:       Handle to the DevNode who's WMD we are querying.
 *      cBufSize:       Size of buffer.
 *      pWMDFileName:   Ptr to a character buffer to hold the WMD filename.
 *  Returns:
 *      DSP_SOK:                On success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_RESOURCENOTAVAIL: The filename is not available.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:        Not more than cBufSize bytes were copied
 *                      into pWMDFileName.
 *
 */
	extern DSP_STATUS CFG_GetWMDFileName(IN struct CFG_DEVNODE* hDevNode,
					     IN ULONG cBufSize,
					     OUT PSTR pWMDFileName);

/*
 *  ======== CFG_GetZLFile ========
 *  Purpose:
 *      Retreive the ZLFile, if any, for this board.
 *  Parameters:
 *      hDevNode:       Handle to the DevNode who's WMD we are querying.
 *      cBufSize:       Size of buffer.
 *      pstrZLFileName: Ptr to character buf to hold ZLFileName.
 *  Returns:
 *      DSP_SOK:                Success.
 *      CFG_E_INVALIDPOINTER:   pstrZLFileName is invalid.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      CFG_E_RESOURCENOTAVAIL: couldn't find the ZLFileName.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    Not more than cBufSize bytes were copied into 
 *                  pstrZLFileName, and *pstrZLFileName contains ZLFileName 
 *                  for this devnode.
 */
	extern DSP_STATUS CFG_GetZLFile(IN struct CFG_DEVNODE* hDevNode,
					IN ULONG cBufSize,
					OUT PSTR pstrZLFileName);

/*
 *  ======== CFG_Init ========
 *  Purpose:
 *      Initialize the CFG module's private state.
 *  Parameters:
 *  Returns:
 *      TRUE if initialized; FALSE if error occured.
 *  Requires:
 *  Ensures:
 *      A requirement for each of the other public CFG functions.
 */
	extern bool CFG_Init();

/*
 *  ======== CFG_SetDevObject ========
 *  Purpose:
 *      Store the Device Object handle for a given devnode.
 *  Parameters:
 *      hDevNode:   Platform's DevNode handle we are storing value with.
 *      dwValue:    Arbitrary value to store.
 *  Returns:
 *      DSP_SOK:                Success.
 *      CFG_E_INVALIDHDEVNODE:  hDevNode is invalid.
 *      DSP_EFAIL:              Internal Error.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:    The Private DWORD was successfully set.
 */
	extern DSP_STATUS CFG_SetDevObject(IN struct CFG_DEVNODE* hDevNode,
					   IN DWORD dwValue);

/*
 *  ======== CFG_SetDrvObject ========
 *  Purpose:
 *      Store the Driver Object handle.
 *  Parameters:
 *      dwValue:        Arbitrary value to store.
 *      dwType          Type of Object to Store
 *  Returns:
 *      DSP_SOK:        Success.
 *      DSP_EFAIL:      Internal Error.
 *  Requires:
 *      CFG initialized.
 *  Ensures:
 *      DSP_SOK:        The Private DWORD was successfully set.
 */
	extern DSP_STATUS CFG_SetObject(IN DWORD dwValue, IN DWORD dwType);

	extern DSP_STATUS CFG_GetC55Procs(OUT DWORD * numProcs);
#ifdef __cplusplus
}
#endif
#endif				/* CFG_ */
